src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldNotMatchKeywords.cs

```diff
@@ -164,7 +165,8 @@ private void AnalyzeNamespaceRule(SymbolAnalysisContext context, HashSet<string>
 
             if (foundKeyword)
             {
-                keywordNamedNamespaces.Add(namespaceDisplayString);
+                keywordNamedNamespaces.AddOrUpdate(namespaceDisplayString, namespaceDisplayString,
+                    (key, oldValue) => namespaceDisplayString);
```

Comment:
According to https://msdn.microsoft.com/en-us/library/dd287191(v=vs.110).aspx#Anchor_7, `TryAdd` is a better choice here.